python/cuml/test/plugins/quick_run_plugin.py

```diff
@@ -0,0 +1,143 @@
+#
+# Copyright (c) 2020, NVIDIA CORPORATION.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+
+import typing
+
+import pytest
+import _pytest.python
+
+
+def pytest_addoption(parser):
+
+    group = parser.getgroup('Quick Run Plugin')
+
+    group.addoption("--quick_run",
+                    default=False,
+                    action="store_true",
+                    help=("Selecting this option will reduce the number of "
+                          "tests run by only picking tests if one of the "
+                          "parameters has never been seen before."))
```

Comment:
I would mention that by doing this, the quick run option stresses code correctness while not running all numeric tests for the algorithms.

Reply:
Fixed. Pending push